central/declarativeconfig/manager_impl.go

```diff
@@ -375,11 +375,14 @@ func (m *managerImpl) removeStaleHealthStatuses(idsToSkip []string) error {
 		return errors.Wrap(err, "retrieving integration health statuses for declarative config")
 	}
 
-	idsToSkipSet := set.NewFrozenStringSet(idsToSkip...)
+	healthIDsToSkip := set.NewStringSet()
+	for _, id := range idsToSkip {
+		healthIDsToSkip.Add(declarativeConfigUtils.DeclarativeConfigHealthIDPrefix + id)
```

Comment:
Since we already have the helper defined within the tests (`addHealthIDPrefix`), can we add a helper within the `declarativeConfigUtils` package that has the following signature:
```go
func IDForIntegrationHealth(id string) string
```

You can either skip changing all the IDs in the existing string set and within the `Contains` call `IDForIntegrationHealth` or remove the current prefix adding.

Also, we can then make `DeclarativeConfigHealthIDPrefix` unexposed, since there's no good reason to have it exposed anymore.